src/Event/PlatformArgumentInjectionEvent.php

```diff
@@ -0,0 +1,183 @@
+   */
+  protected $command;
+  
+  /**
+   * Decorated input containing the mapping between sites and their args.
+   * 
+   * @var array
+   */
+  protected $decoratedInput;
+
+  /**
+   * PlatformArgumentInjectionEvent constructor.
+   *
+   * @param \Symfony\Component\Console\Input\InputInterface $input
+   *   The current command's input object.
+   * @param array $sites
+   *   The list of sites.
+   * @param \Symfony\Component\Console\Command\Command $command
+   *   The current command.
+   */
+  public function __construct(InputInterface $input, array $sites, Command $command) {
+    $this->input = $input;
+    $this->sites = $sites;
+    $this->command = $command;
+  }
+
+  /**
+   * The available sites for the given platform.
+   * 
+   * @return array
+   *   The list of sites. Should be used as key.
+   */
+  public function getSites(): array {
+    return $this->sites;
+  }
+
+  /**
+   * Returns the name of the current command.
+   *
+   * @return string
+   *   The name of the command.
+   */
+  public function getCommandName(): string {
+    return $this->command->getName();
+  }
+
+  /**
+   * Returns the input in hand.
+   *
+   * @return \Symfony\Component\Console\Input\InputInterface
+   *   The current command's input object.
+   */
+  public function getInput(): InputInterface {
+    return $this->input;
+  }
+
+  /**
+   * Returns the decorated input.
+   * 
+   * @return array
+   *   The list of inputs mapped to platform sites.
+   */
+  public function getDecoratedInput(): array {
+    return $this->decoratedInput;
+  }
+
+  /**
+   * Maps customized arguments to sites derived from a platform.
+   *
+   * Example:
+   * ```php
+   *   [
+   *     'site_name' => [
+   *       '--option' => 'option_value',
+   *       'some_argument' => 'arg_value',
+   *     ],
+   *   ]
+   * ```
+   *
+   * @param array $arguments
+   *   An associative array containing the sites and the customized input.
+   *
+   * @throws \Exception
+   */
+  public function setDecoratedInput(array $arguments): void {
+    $args = $this->input->getArguments();
+    $options = $this->input->getOptions();
+    $definition = $this->command->getDefinition();
+    foreach ($options as $key => $val) {
+      $option = $definition->getOption($key);
+      // Option value must be TRUE so it can be qualified as a used option.
+      if ($option->acceptValue() && (bool) $val === TRUE) {
+        $options["--$key"] = $val;
+      }
+      else {
+        // Insert the option into the array IF it was actually used.
+        if ((bool) $val === TRUE) {
+          $options["--$key"] = NULL;
+        }
+      }
+
+      unset($options[$key]);
+    }
+
+    $res = [];
+    foreach ($arguments as $site => $params) {
+      if (!in_array($site, $this->sites, TRUE)) {
+        throw new \Exception("No such site: $site");
+      }
+      
+      if ($params !== NULL) {
+        foreach ($params as $arg => $val) {
+          if ($this->isOption($arg)) {
```

Comment:
Shouldn't we check that the arguments passed here are valid for the option they're being mapped to? Like if it doesn't accept a value, but one was passed, let's just set it up properly with the null value.